### PR TITLE
ES6: Convert sport/score-board

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `sport/score-board.js` to ES6.

## What is the value of this and can you measure success?

ES6

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/sport/score-board.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        this.render(this.placeholder);
+    }
+
+    error(): void {
+        this.placeholder.innerHTML = '';
+        this.parent.removeClass('u-h');
+    }
+}
+
+export default ScoreBoard;
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/sport/score-board.js]: Why not a named export?

File: static/src/javascripts/projects/common/modules/sport/score-board.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <div class="score__loading ${context.loadingState}">
+            <div class="loading__text">Fetching the scores…</div>
+            <div class="is-updating"></div>
+        </div>
+    </div>
+`;
+
+class ScoreBoard extends Component {
+    constructor(context: Object): void {
+        super(context);
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/sport/score-board.js]: Is it necessary to pass `context` to the Component's constructor?  The parent constructor doesn't accept any arguments or contain any logic.

File: static/src/javascripts/projects/common/modules/sport/score-board.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        const scoreContainerHtml = getScoreContainerHtml({
+            loadingState: this.pageType !== 'report'
+                ? ' score__loading--live'
+                : '',
+        });
+
+        this.componentClass = 'match-summary';
+        this.pageType = context.pageType;
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/sport/score-board.js]: If we're assigning all the properties of `context` to `this`we could replace lines 28-32 with `Object.assign(this, context)`...

File: static/src/javascripts/projects/common/modules/sport/score-board.js (hunk 2 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/sport/score-board.js]: Good point. Can already hear the giggle of @SiAdcock behind my back ;)

File: static/src/javascripts/projects/common/modules/sport/score-board.js (hunk 3 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/sport/score-board.js]: You're right. I should then probably add type definitions for `context` ...

**Comment by gustavpursche** [comment]
@GHaberis @SiAdcock Thanks for the review! Changed everything, so if you like you could give it another 👀 .

**Review by SiAdcock — APPROVED** [review]
❤️

**Review by GHaberis — APPROVED** [review]
Nice :sunglasses:

**Pull request merged**